astropy/utils/report_info.py

```diff
@@ -0,0 +1,98 @@
+r"""
+A helper script to automate system info in bug reports.
+
+This can be run in two fashions either by downloading the latest version of this
+script and running it on the fly
+```
+curl --silent https://raw.githubusercontent.com/astropy/astropy/main/astropy/utils/report_info.py | python
+```
+
+or, from a local install
+```
+python -c "from astropy.utils.report_info import main; main()"
+```
+
+Running this script should not require anything else than the standard library.
+"""
+
+import platform
+from importlib.metadata import version
+from importlib.util import find_spec
+from itertools import chain
+
+
+def header(name):
+    return [name, "-" * len(name)]
```

Comment:
Adding type annotations is not mandatory, but I recommend adding them nonetheless:
```suggestion
def header(name: str) -> tuple[str, str]:
    return name, len(name) * "-"
```
or perhaps
```suggestion
def header(name: str) -> str:
    return f"{name}\n{len(name) * '-'}"
```

Reply:
I'd like to keep the return type a list for two reasons:
- in tuples, position is significant (borrowing the "Cultural Difference between lists and tuples" from https://nedbatchelder.com/blog/201608/lists_vs_tuples.html)
- your second suggestion requires more changes

Reply:
I'd prefer to keep the output immutable unless there is good reason to make it mutable, but these functions should never be called outside this module, so it doesn't really matter.